pgcli/pgcompleter.py

```diff
@@ -600,23 +633,76 @@ def get_from_clause_item_matches(self, suggestion, word_before_cursor):
         s = suggestion
         t_sug = Table(s.schema, s.table_refs, s.local_tables)
         v_sug = View(s.schema, s.table_refs)
-        f_sug = Function(s.schema, s.table_refs, filter='for_from_clause')
+        f_sug = Function(s.schema, s.table_refs, usage='from')
         return (self.get_table_matches(t_sug, word_before_cursor, alias)
             + self.get_view_matches(v_sug, word_before_cursor, alias)
             + self.get_function_matches(f_sug, word_before_cursor, alias))
 
-    # Note: tbl is a SchemaObject
-    def _make_cand(self, tbl, do_alias, suggestion):
+    def _arg_list(self, func, usage):
+        """
+        Returns a an arg list string, e.g. `(_foo:=23, bar:='baz')` for a func
+        func is a FunctionMetadata object
+        usage is 'call', 'call_display' or 'signature'
+        """
+        template = {
+            'call':  self.call_arg_style,
+            'call_display': self.call_arg_display_style,
+            'signature': self.signature_arg_style
+        }[usage]
+        args = func.args()
+        if not template:
+            return '()'
+        multiline = usage == 'call' and len(args) > self.call_arg_oneliner_max
+        max_arg_len = max(len(a.name) for a in args) if multiline else 0
+        args = (
+            self._format_arg(template, arg, arg_num + 1, max_arg_len)
+            for arg_num, arg in enumerate(args)
+        )
+        if multiline:
+            return '(' + ','.join('\n    ' + a for a in args if a) + '\n)'
+        else:
+            return '(' + ', '.join(a for a in args if a) + ')'
+
+    def _format_arg(self, template, arg, arg_num, max_arg_len):
+        if not template:
+            return None
+        if arg.has_default:
+            arg_default = 'NULL' if arg.default is None else arg.default
+            # Remove trailing ::(schema.)type
+            arg_default = re.sub(r'::[\w\.]+(\[\])?$', '', arg_default)
```

Comment:
The regex could be made a `re.compile`d module level global so it only needs to be compiled once